### ISSUE TITLE
The enable service security feature is not working in some environment due to the iptables-legacy executable not being found

### ISSUE THREAD

**Issue by starbops** [kube-vip/kube-vip#808]
**Describe the bug**

When having a kube-vip setup with an `enable_service_security=true` config, creating new LB type of services results in the "iptables-legacy executable not found" log in the kube-vip pod. Hence kube-vip cannot install the relevant iptables rules to limit the traffic other than the specified ones.

```shell
$ kubectl -n harvester-system logs kube-vip-mgdgr                                                                       
time="2024-04-08T05:22:51Z" level=info msg="Starting kube-vip.io [v0.7.2]"
time="2024-04-08T05:22:51Z" level=info msg="namespace [kube-system], Mode: [ARP], Features(s): Control Plane:[false], Services:[true]"
time="2024-04-08T05:22:51Z" level=info msg="No interface is specified for VIP in config, auto-detecting default Interface"
time="2024-04-08T05:22:51Z" level=info msg="prometheus HTTP server started"
time="2024-04-08T05:22:51Z" level=info msg="kube-vip will bind to interface [mgmt-br]"
time="2024-04-08T05:22:51Z" level=info msg="Starting Kube-vip Manager with the ARP engine"
time="2024-04-08T05:22:51Z" level=info msg="beginning services leadership, namespace [harvester-system], lock name [plndr-svcs-lock], id [harvester-qjl6m]"
I0408 05:22:51.367476       1 leaderelection.go:250] attempting to acquire leader lease harvester-system/plndr-svcs-lock...
I0408 05:22:51.375358       1 leaderelection.go:260] successfully acquired lease harvester-system/plndr-svcs-lock
time="2024-04-08T05:22:51Z" level=info msg="(svcs) starting services watcher for all namespaces"
time="2024-04-08T05:22:51Z" level=info msg="(svcs) adding VIP [[192.168.48.240]] for [kube-system/ingress-expose]"
time="2024-04-08T05:22:51Z" level=info msg="[service] synchronised in 4ms"
time="2024-04-08T05:22:51Z" level=info msg="(svcs) adding VIP [[192.168.48.51]] for [default/test-vm-lb]"
time="2024-04-08T05:22:51Z" level=warning msg="could not add iptables rules to limit traffic ports: could not create iptables client: exec: \"iptables-legacy\": executable file not found in $PATH"
time="2024-04-08T05:22:51Z" level=info msg="[service] synchronised in 2ms"
```

The `iptables-legacy` symlink is no longer there because we bumped the Alpine version to v3.19 since kube-vip v0.7.0:

https://github.com/kube-vip/kube-vip/blob/1ea277f43e76a27894a83e305c880b68cf5c5f31/Dockerfile_iptables#L14-L17

According to the [Alpine release notes](https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.19.0), now it defaults to `xtables-nft-multi` instead of `xtables-legacy-multi`.

To be backward compatible, we should consider installing `iptables-legacy` explicitly:

```Dockerfile
FROM alpine:3.19.1
# Update pkgs and add iptables
RUN apk upgrade && \
    apk add --no-cache iptables iptables-legacy
```

**To Reproduce**

1. Install Harvester v1.2.1
2. Create a VM Network
3. Create a VM
4. Create an IP pool
5. Create a load balancer pointing to the VM with a specific port, e.g. port 80 for HTTP
6. Try to access the VM with port 22 using `ssh` and succeed. But it should be prohibited because we only specified port 80 for load balancing.

**Expected behavior**

The "Enable Service Security" feature should work correctly on a system with legacy iptables.

**Screenshots**

N/A

**Environment (please complete the following information):**
 - OS/Distro: SLE Micro 5.3 (Harvester v1.2.1)
   ```shell
   $ uname -a
   Linux harvester-qjl6m 5.14.21-150400.24.92-default #1 SMP PREEMPT_DYNAMIC Wed Oct 4 16:07:15 UTC 2023 (3731029) x86_64 x86_64 x86_64 GNU/Linux
   ```
 - Kubernetes Version: v1.25.9 (v1.25.9+rke2r1)
 - Kube-vip Version: 0.7.2

**`Kube-vip.yaml`:**

```yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  annotations:
    deprecated.daemonset.template.generation: "4"
    meta.helm.sh/release-name: harvester
    meta.helm.sh/release-namespace: harvester-system
    objectset.rio.cattle.io/id: default-mcc-harvester-cattle-fleet-local-system
  creationTimestamp: "2024-04-01T04:17:53Z"
  generation: 4
  labels:
    app.kubernetes.io/managed-by: Helm
    objectset.rio.cattle.io/hash: e852fa897f5eae59a44b4bfe186aad80b10b94b3
  name: kube-vip
  namespace: harvester-system
  resourceVersion: "1260845"
  uid: 6afcdc4f-c332-4ef6-89e1-740a4625635d
spec:
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app.kubernetes.io/instance: harvester
      app.kubernetes.io/name: kube-vip
  template:
    metadata:
      creationTimestamp: null
      labels:
        app.kubernetes.io/instance: harvester
        app.kubernetes.io/name: kube-vip
    spec:
      containers:
      - args:
        - manager
        env:
        - name: cp_enable
          value: "false"
        - name: enable_service_security
          value: "true"
        - name: lb_enable
          value: "true"
        - name: lb_port
          value: "6443"
        - name: svc_enable
          value: "true"
        - name: vip_arp
          value: "true"
        - name: vip_cidr
          value: "32"
        - name: vip_interface
        - name: vip_leaderelection
          value: "false"
        image: ghcr.io/kube-vip/kube-vip-iptables:v0.7.2
        imagePullPolicy: IfNotPresent
        name: kube-vip
        resources: {}
        securityContext:
          capabilities:
            add:
            - NET_ADMIN
            - NET_RAW
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      hostNetwork: true
      nodeSelector:
        node-role.kubernetes.io/control-plane: "true"
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: kube-vip
      serviceAccountName: kube-vip
      terminationGracePeriodSeconds: 30
      tolerations:
      - effect: NoSchedule
        key: node-role.kubernetes.io/control-plane
        operator: Exists
  updateStrategy:
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 1
    type: RollingUpdate
status:
  currentNumberScheduled: 1
  desiredNumberScheduled: 1
  numberAvailable: 1
  numberMisscheduled: 0
  numberReady: 1
  observedGeneration: 4
  updatedNumberScheduled: 1
```

**Additional context**

We try to upgrade kube-vip from v0.6.0 to v0.7.2 but encountered this issue. It used to be working fine in v0.6.0.

**Comment by thebsdbox** [comment]
Merged!

**Comment by starbops** [comment]
Hi @thebsdbox Could you let us know when the new version is expected to be released? Thanks!

**Comment by thebsdbox** [comment]
Fixed in v0.8.0

**Comment by aaronps** [comment]
I just had this error using v0.8.7, I removed enable_service_security=true and kube-vip started successfully.

"Fixed in v0.8.0" seems misleading.

**Comment by luisbozz** [comment]
> I just had this error using v0.8.7, I removed enable_service_security=true and kube-vip started successfully.

Same here on v0.8.9
